pycoshark/mongomodels.py

```diff
@@ -616,32 +460,200 @@ class PullRequestFile(Document):
     patch = StringField()
 
 
-class IssueSystem(Document):
+class ActionWorkflow(Document):
     """
-    IssueSystem class.
-    Inherits from :class:`mongoengine.Document`
+    `Represents a workflow in a collection.
+
+    This class inherits from :class:`~mongoengine.Document` and defines the schema for a workflow.
+
+    :property workflow_id: (:class:`~mongoengine.fields.IntField`) Unique identifier for the workflow.
+    :property name: (:class:`~mongoengine.fields.StringField`) Name of the workflow.
+    :property path: (:class:`~mongoengine.fields.StringField`) Path of the workflow (default is None).
+    :property state: (:class:`~mongoengine.fields.StringField`) Current state of the workflow.
+    :property created_at: (:class:`~mongoengine.fields.DateTimeField`) Date and time when the workflow was created.
+    :property updated_at: (:class:`~mongoengine.fields.DateTimeField`) Date and time when the workflow was last updated.
+    :property project_url: (:class:`~mongoengine.fields.StringField`) URL of the associated project (default is None).
+    :property project_id: (:class:`~mongoengine.fields.ObjectIdField`) ID of the associated project (default is None).
+    :property vcs_system_id: (:class:`~mongoengine.fields.ObjectIdField`) ID of the associated VCS system (default is None).`
+    """
+    meta = {
+        'indexes': [
+            'ci_system_ids',
+            'external_id'
+        ]
+    }
+    ci_system_ids = ListField(required=True)
+    external_id = StringField()
+    name = StringField()
+    path = StringField(default=None)
+    state = StringField()
+    created_at = DateTimeField()
+    updated_at = DateTimeField()
+    project_url = StringField(default=None)
+    project_id = ObjectIdField(default=None)
+    vcs_system_id = ObjectIdField(default=None)
```

Comment:
Without a run collection between system and the data collections we have no way to express a singular vcs_system_id which represents the repository. We should think about whether we want to have only the URL in those cases or leave it like it is in the next meeting.